core/domain/user_jobs_one_off.py

```diff
@@ -94,26 +94,38 @@ def entity_classes_to_map_over(cls):
     @staticmethod
     def map(item):
         """Implements the map function for this job."""
-        user_id = item.id
-        gae_id = [item.gae_id]
-        user_auth_model = user_models.UserAuthModel.get_by_auth_id(
-            feconf.AUTH_METHOD_GAE, gae_id[0])
-
-        # For current migration, one user_id will strictly have one gae_id
-        # associated with it and vice versa.
-        if user_auth_model is None:
-            user_models.UserAuthModel(
-                id=user_id,
-                gae_id=gae_id
-            ).put()
-        elif user_auth_model.id != user_id:
-            # Because <user_id, gae_id> tuple coming from UserSettingsModel
-            # currently is the source of truth.
-            user_models.UserAuthModel.delete_by_id(user_auth_model.id)
-            user_models.UserAuthModel(
-                id=user_id,
-                gae_id=gae_id
-            ).put()
+
+        # Assumption: full user_id + all its associated profile user_ids will
```

Comment:
This doesn't quite make sense to me. How will GAE login work in this case? Isn't the expectation that you log into Oppia via a Google account through app engine, receive a GAE ID, then lookup the user to log into? How do we know which one to pick if there's a 1:many relationship from GAE ID to user ID?

Per the other comment thread, I was expecting that profiles would authenticate using a user ID + PIN, not a GAE ID. The idea there is that a profile logs in by providing a PIN and a valid user ID that's already been logged in via some other method (such as GAE ID).

It's also the case that there will be profiles associated with accounts that do not use GAE to auth (such as phone numbers).

Reply:
Ok, now I get what you meant by storing PIN in `UserAuthModel`. Please confirm if I am understanding it correctly: 

In the UserAuthModel we have:
- full user: user_id, gae_id, pin(maybe None)
- profile user: user_id, full_user_id, pin(maybe None)

Post-authentication flow in the backend:
- a full user sends gae_id in the request - we lookup by gae_id in the UserAuthModel
- a profile user sends its own user_id + pin + full user in the request - correct full user id assures us that the user has been authenticated(right? This is the part I am not sure of) i.e existence of <full_user_id, user_id, pin> entry in the UserAuthModel.

Reply:
That's close to what I was thinking, but I just want to clarify a couple things:
1. Are you planning to keep 'gae_id' and the parent 'full_user_id' distinct?
2. For authenticating profiles, everything you mentioned matches what I was thinking except we need to verify that the parent full user ID is correctly authenticated. My suggestion is to not do any additional storage work to accommodate this--we need to figure this out during the authentication project (@anandwana001 FYI).

Reply:
1) Yes gae_id different from full_user_id. Following table shows the intent:

| Attribute                                              | Full user | Profile user |
|--------------------------------------------------------|-----------|--------------|
| user_id (instances of this class are keyed by user_id) | Yes       | Yes          |
| gae_id                                                 | Yes       | None         |
| PIN (optional for both)                                | Yes       | Yes          |
| parent_user_id                                         | None      | Yes          |

None of the above attributes are **required** (except user_id which will be the `id`attribute). Whether they should be None or not will be ensured by external validation.

2) Okay, as I will keep it to this much only.